### PR TITLE
Added node type utils

### PR DESCRIPTION
**Pull request by Cisplatin**
As discussed in https://github.com/duaraghav8/solium-plugin-security/pull/24, given that other plugins besides https://github.com/duaraghav8/solium-plugin-security will be using these utill functions, it was suggested by @duaraghav8 that we centralize the file in Solium. This PR does that.

### CODE REVIEW

File: lib/utils/ast-utils.js [hunk 1]
```diff
@@ -35,6 +35,79 @@ exports.init = function (sourceCode) {
  */
 exports.isASTNode = nodeSchemaValidator.compile (astNodeSchema);
 
+/**
+ * @param {Object} node The node to check
+ * @param {String} type The type of the node
+ * @returns {Boolean} true if the given node is the right type
+ */
+function isNodeType (node, name) {
```
Comment (duaraghav8) [inline lib/utils/ast-utils.js]: Please add a validation at the start: `throwIfInvalidNode(node, '<function name>');`. Method is already defined, just need to use it.
See [example](https://github.com/duaraghav8/Solium/blob/master/lib/utils/ast-utils.js#L44).

File: test/lib/utils/source-code-utils.js [hunk 2]
```diff
@@ -42,6 +42,31 @@ describe ('Testing SourceCode instance for exposed functionality', function () {
 		sourceCodeObject.should.have.property ('getParent');
 		sourceCodeObject.getParent.should.be.type ('function');
 
+		// functions for node type checking
```
Comment (duaraghav8) [inline test/lib/utils/source-code-utils.js]: Please add cases to test the functionality of your util methods too. You need only have 2 cases per method:
1 that is expected to return `true` & another expected to return `false`.

File: lib/utils/ast-utils.js (hunk 1 above)
Comment (duaraghav8) [reply lib/utils/ast-utils.js]: since all your util methods call `isNodeType`, only this function needs to call this validation method.

**Comment by duaraghav8** [comment]
@Cisplatin 2 small additions and then we're solid!

**Comment by Cisplatin** [comment]
@duaraghav8 Added those two things you asked for

**Pull request merged**